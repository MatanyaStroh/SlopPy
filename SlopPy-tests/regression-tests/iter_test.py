# iterators should NOT yield NA objects

x = 1 / 0
assert type(x) is NA

# list
y = [1, 2, x, 3, 4]
z = []
for e in y:
  z.append(e)

assert z == [1, 2, 3, 4]

# tuple:
y = (1, 2, x, 3, 4, x)
z = []
for e in y:
  z.append(e)

assert z == [1, 2, 3, 4]

# set:
y = set()
y.add(1)
y.add(2)
y.add(x)
y.add(3)
y.add(4)

z = set()
for e in y:
  z.add(e)

assert z == set([1, 2, 3, 4])

