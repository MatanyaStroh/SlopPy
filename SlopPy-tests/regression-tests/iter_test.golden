{NA <type 'exceptions.ZeroDivisionError'> 'integer division or modulo by zero'
 Traceback (most recent call last):
  File "regression-tests/iter_test.py", line 3, in <module>
    x = 1 / 0

Locals: {'__builtins__': <module '__builtin__' (built-in)>, '__name__': '__main__', '__file__': 'regression-tests/iter_test.py', '__doc__': None, '__package__': None}
}
{iterator skipped over NA | regression-tests/iter_test.py <module>:9}
{iterator skipped over NA | regression-tests/iter_test.py <module>:17}
{iterator skipped over NA | regression-tests/iter_test.py <module>:17}
{iterator skipped over NA | regression-tests/iter_test.py <module>:31}
